src/main/java/org/rumbledb/runtime/functions/sequences/general/SubsequenceFunctionIterator.java

```diff
@@ -79,6 +86,53 @@ public SubsequenceFunctionIterator(
         return SparkSessionManager.getInstance().getJavaSparkContext().emptyRDD();
     }
 
+    @Override
+    protected boolean implementsDataFrames() {
+        return true;
+    }
+
+    @Override
+    public Dataset<Row> getDataFrame(DynamicContext dynamicContext) {
+        Dataset<Row> df = this.sequenceIterator.getDataFrame(dynamicContext);
+        setInstanceVariables(dynamicContext);
+        StructType inputSchema = df.schema();
+
+        List<String> allColumns = FlworDataFrameUtils.getColumnNames(inputSchema, -1, null);
+
+        String selectSQL = FlworDataFrameUtils.getSQL(allColumns, true);
+
+        df.createOrReplaceTempView("input");
+        df = df.sparkSession()
+            .sql(
+                String.format(
+                    "SELECT * FROM input LIMIT %s",
+                    Integer.toString(this.startPosition + this.length - 1)
+                )
+            );
+
+        df = FlworDataFrameUtils.zipWithIndex(df, 1L, "foo");
```

Comment:
"foo" is very likely to collide with a user column. Can we use a UUID for the column name here?

Reply:
Ah, yes! That was the plan and I forgot. Thanks! Very good eyes.